packages/eslint-plugin/src/util/isArrayMethodCallWithPredicate.ts

```diff
@@ -30,7 +30,10 @@ export function isArrayMethodCallWithPredicate(
 
   const staticAccessValue = getStaticMemberAccessValue(node.callee, context);
 
-  if (!staticAccessValue || !ARRAY_PREDICATE_FUNCTIONS.has(staticAccessValue)) {
+  if (
+    typeof staticAccessValue !== 'string' ||
+    !ARRAY_PREDICATE_FUNCTIONS.has(staticAccessValue)
+  ) {
```

Comment:
(nit) I think applying DeMorgan on these makes the intention much more clear, but up to you! 🙂 
```suggestion
  if (
    !(typeof staticAccessValue === 'string' &&
       ARRAY_PREDICATE_FUNCTIONS.has(staticAccessValue)
    )
  ) {
```

Reply:
(side note) - I _really_ wish that `Set#has` were typed to allow `unknown` rather than the type of the set . Having to narrow the type to the set is often counterproductive. 

I personally actually really wouldn't mind doing
```suggestion
  if (
    ARRAY_PREDICATE_FUNCTIONS.has(staticAccessValue as any)
  ) {
```
but i have a feeling that might not be so popular

Reply:
I 200% agree with you about the type of `Set#has` (and `Array#includes`) haha...
https://github.com/mattpocock/ts-reset